src/main/java/reposense/model/AuthorConfiguration.java

```diff
@@ -0,0 +1,205 @@
+    private static final Logger logger = LogsManager.getLogger(AuthorConfiguration.class);
+
+    private RepoLocation location;
+    private String branch;
+    private String displayName;
+
+    private transient List<Author> authorList = new ArrayList<>();
+    private transient TreeMap<String, Author> authorEmailsAndAliasesMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    private transient Map<Author, String> authorDisplayNameMap = new HashMap<>();
+
+    public AuthorConfiguration(RepoLocation location) {
+        this(location, DEFAULT_BRANCH);
+    }
+
+    public AuthorConfiguration(RepoLocation location, String branch) {
+        this.location = location;
+        this.branch = location.isEmpty() ? DEFAULT_BRANCH : branch;
+
+        String organization = location.getOrganization();
+        String repoName = location.getRepoName();
+
+        if (organization != null) {
+            displayName = organization + "_" + repoName + "_" + branch;
+        } else {
+            displayName = repoName + "_" + branch;
+        }
+    }
+
+    /**
+     * Clears authors information and use the information provided from {@code standaloneConfig}.
+     */
+    public void update(StandaloneConfig standaloneConfig) {
+        List<Author> newAuthorList = new ArrayList<>();
+        TreeMap<String, Author> newAuthorEmailsAndAliasesMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        Map<Author, String> newAuthorDisplayNameMap = new HashMap<>();
+        List<String> newIgnoreGlobList = standaloneConfig.getIgnoreGlobList();
+
+        for (StandaloneAuthor sa : standaloneConfig.getAuthors()) {
+            Author author = new Author(sa);
+            author.appendIgnoreGlobList(newIgnoreGlobList);
+
+            newAuthorList.add(author);
+            newAuthorDisplayNameMap.put(author, author.getDisplayName());
+            List<String> aliases = new ArrayList<>(author.getAuthorAliases());
+            List<String> emails = new ArrayList<>(author.getEmails());
+            aliases.add(author.getGitId());
+            aliases.forEach(alias -> newAuthorEmailsAndAliasesMap.put(alias, author));
+            emails.forEach(email -> newAuthorEmailsAndAliasesMap.put(email, author));
+        }
+
+        Format.validateFormats(standaloneConfig.getFormats());
+        CommitHash.validateCommits(standaloneConfig.getIgnoreCommitList());
+
+        // only assign the new values when all the fields in {@code standaloneConfig} pass the validations.
+        setAuthorList(newAuthorList);
+        setAuthorEmailsAndAliasesMap(newAuthorEmailsAndAliasesMap);
+        setAuthorDisplayNameMap(newAuthorDisplayNameMap);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof AuthorConfiguration)) {
+            return false;
+        }
+
+        return hashCode() == ((AuthorConfiguration) other).hashCode();
```

Comment:
We can make `AuthorConfiguration` check for all of its value (instead of just location and branch, which is only used in the step to find the correct `RepoConfiguration` mapping)?